go.mod

```diff
@@ -1,23 +1,23 @@
 module github.com/operator-framework/api
 
-go 1.17
+go 1.18
```

Comment:
I do not think that we should upgrade here. 
That will force all consumers use go 1.18
WDYT about we make all projects support go 1.18 first?

Reply:
That is a good point.

I was thinking it would be best to try and do the k8s 1.24 and Go 1.18 bump at the same time since k8s 1.24 is built with Go 1.18, but now that you point it out it may not be as easy for other projects to make the Go 1.18 bump. 

I think it may be best to split it into 2 separate bumps - one for k8s 1.24 and then one for Go 1.18. I will go ahead and update this. 

Thanks for pointing this out @camilamacedo86